lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3707,3 +3733,75 @@ def get_test_data(delta=0.05):
     Y = Y * 10
     Z = Z * 500
     return X, Y, Z
+
+
+class _Quaternion:
+    """
+    Quaternions
+    consisting of scalar, along 1, and vector, with components along i, j, k
+    """
+
+    def __init__(self, scalar, vector):
+        self.scalar = scalar
+        self.vector = np.array(vector)
+
+    def __neg__(self):
+        return self.__class__(-self.scalar, -self.vector)
+
+    def __mul__(self, other):
+        """
+        Product of two quaternions
+        i*i = j*j = k*k = i*j*k = -1
+        """
+        return self.__class__(
+            self.scalar*other.scalar - np.dot(self.vector, other.vector),
+            self.scalar*other.vector + self.vector*other.scalar
+            + np.cross(self.vector, other.vector))
```

Comment:
This should be unit normed so that we do not accumulate floating point inaccuracies.

Reply:
There is no need, the `as_cardan_angles()` and `from_cardan_angles()` take care of that (indeed there is a test for it).

In case anyone would need normalizing, I'd strongly suggest to introduce a method for that specifically ("`q.normalize()`").

Also, the name of the class suggests that it is general - if the class dealt with normalized quaternions only, then wouldn't you have expected `_NormalizedQuaternion`, or `_Versor`, as a class name?

Reply:
> My broader worry is that someone down the road will purposefully or accidentally create non-normed quaternions with this class and then `as_cardan_angles` will give nonsense results.

`as_cardan_angles()` is indifferent to the magnritude of the quaternion, it gives sensible results, no worries.

> Another way to think about the question - is there ever going to be a use for _non_-unit-norm quaternions here?

Of course. It can used to represent scaling of the axes as well.